### PR TITLE
infoschema: version of tiflash is inconsistent with other component in table information_schema.cluster_info

### LINKED ISSUE #24693: infoschema: version of tiflash is inconsistent with other component in table information_schema.cluster_info
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Set up a cluster with tiflash and execute query 
```
select * from information_schema.cluster_info;
```

### 2. What did you expect to see? (Required)

image

### 3. What did you see instead (Required)

version of tiflash should be `4.0.12` not `v4.0.12` 

### 4. What is your TiDB version? (Required)

4.0.12, but that is the same with master branch

### PR DESCRIPTION
**Pull request by yuqi1129**
### What problem does this PR solve?

Issue Number: close #24693 

Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:
if version of tiflash  start with 'v', we remove it,  like 
```
f := func(v string) string {
		if v[0] == 'v' {
			v = v[1:]
		}

		return v
	}
```

How it Works:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

- No release note

### CODE REVIEW

**Comment by yuqi1129** [comment]
Can anyone help to view this PR?

File: infoschema/tables_test.go [hunk 1]
```diff
@@ -956,6 +956,22 @@ func (s *testTableSuite) TestFormatVersion(c *C) {
...
+	versions = []string{"v4.0.12", "4.0.12", "v5.0.1"}
+	resultVersion := []string{"4.0.12", "4.0.12", "5.0.1"}
+
+	f := func(v string) string {
+		if len(v) >= 1 && v[0] == 'v' {
+			v = v[1:]
+		}
+
+		return v
+	}
```
Comment (djshow832) [inline infoschema/tables_test.go:969]: Well done!
But I think it's not a good idea to copy the code block. Maybe you can create a function like `FormatTiFlashVersion`, just like what `FormatVersion` does?
Besides, I think `FormatVersion` needs to be renamed to `FormatTiDBVersion`.

Comment (yuqi1129) [reply infoschema/tables_test.go:969]: Indeed, it's more elegant, thanks for your suggestion

File: infoschema/tables.go [hunk 2]
```diff
@@ -1662,18 +1662,26 @@ func GetStoreServerInfo(ctx sessionctx.Context) ([]ServerInfo, error) {
...
-			Version:        store.Version,
+			Version:        FormatTiflashVersion(store.Version),
 			GitHash:        store.GitHash,
 			StartTimestamp: store.StartTimestamp,
 		})
 	}
 	return servers, nil
 }
 
+func FormatTiflashVersion(version string) string {
```
Comment (xhebox) [inline infoschema/tables.go:1678]: It is actually `FormatTiKVAndTiFlashVersion`, right? BTW, please add a comment for the exported method.

Comment (yuqi1129) [reply infoschema/tables.go:1678]: > It is actually FormatTiKVAndTiFlashVersion, right

Good catch, Yes.

> please add a comment for the exported method.

thanks for your advice

Comment (xhebox) [reply infoschema/tables.go:1678]: > thanks for your advice

It actually will the CI test:

```
[2021-06-17T03:30:58.992Z] GO111MODULE=on go build -o ../bin/revive github.com/mgechev/revive

[2021-06-17T03:30:58.992Z] linting

[2021-06-17T03:30:58.992Z]   ✘  https://revive.run/r#exported  exported function FormatTiflashVersion should have comment or be unexported  

[2021-06-17T03:30:58.992Z]   infoschema/tables.go:1678:1

[2021-06-17T03:30:58.992Z] 

[2021-06-17T03:30:58.992Z] ✘ 1 problem (1 error, 0 warnings)

[2021-06-17T03:30:58.992Z] 

[2021-06-17T03:30:58.992Z] Errors:

[2021-06-17T03:30:58.992Z]   1  exported  
```

**Review by xhebox — APPROVED** [review]

**Review by djshow832 — APPROVED** [review]

**Pull request merged**